geetools/ImageCollection.py

```diff
@@ -1740,3 +1740,102 @@ def to_int(d):
+            geometry: The region over which to reduce the data.
+            scale: A nominal scale in meters to work in.
+            crs: The projection to work in. If unspecified, the projection of the image's first band is used. If specified in addition to scale, rescaled to the specified scale.
+            crstransform: The list of CRS transform values. This is a row-major ordering of the 3x2 transform matrix. This option is mutually exclusive with 'scale', and replaces any transform already set on the projection.
+            bestEffort: If the polygon would contain too many pixels at the given scale, compute and use a larger scale which would allow the operation to succeed.
+            maxPixels: The maximum number of pixels to reduce.
+            tileScale: A scaling factor between 0.1 and 16 used to adjust aggregation tile size; setting a larger tileScale (e.g., 2 or 4) uses smaller tiles and may enable computations that run out of memory with the default.
+
+        Returns:
+            A dictionary with the reduced values for each image.
+
+        Examples:
+            .. code-block:: python
+
+                import ee, geetools
+
+                ee.Initialize()
+
+                collection = (
+                    ee.ImageCollection("LANDSAT/LC08/C01/T1_TOA")
+                    .filterBounds(ee.Geometry.Point(-122.262, 37.8719))
+                    .filterDate("2014-01-01", "2014-12-31")
+                )
+                data = collection.geetools.reduceRegion("mean", geometry=ee.Geometry.Point(-122.262, 37.8719), scale=30)
+                print(data.getInfo())
+        """
+        # The most critical part is parsing the idProperty to transform it into list of string compatible
+        # with band names and do it server-side. The 3 cases that we take into account are:
+        # String, Number, Date. The last two are transformed into string.
+        propertyList = self._obj.aggregate_array(idProperty)
+        if idPropertyType == ee.String:
+            propertyList = propertyList.map(lambda p: ee.String(p))
+        elif idPropertyType == ee.Number:
+            propertyList = propertyList.map(lambda p: ee.Number(p).format())
+        elif idPropertyType == ee.Date:
+            propertyList = propertyList.map(lambda p: ee.Date(p).format(EE_DATE_FORMAT))
+        else:
+            raise ValueError("idPropertyType format {idPropertyType} not supported (yet)!")
+
+        # The tobands method will produce an image with the following band names: <system:index>_<bandName>
+        # What we want is: <idProperty>_<bandName> so we can make more advance filtering downstream.
+        bands = self._obj.first().bandNames()
+        bandNames = propertyList.map(lambda p: bands.map(lambda b: ee.String(p).cat("_").cat(b)))
+        bandNames = bandNames.flatten()
+
```

Comment:
``` suggestion
if geometry is None:
    fc = ee.FeatureCollection(self._obj.map(lambda i: ee.Feature(i.geometry()))
    geometry = fc.geometry().dissolve(1)  # should we allow the user to set error margin?
```
due to https://code.earthengine.google.com/0e2501747ea77c19bd7ad130a0cc5fe4

Reply:
This an hinting error, the geometry parameter is compulsory in the image reduceRegion it should be compulsory here as well.

Reply:
solved in https://github.com/gee-community/geetools/pull/359/commits/7fd2e53152d56f04c11fd13dd17516e7067a626f